src/SonarScanner.MSBuild.PreProcessor/SonarServer.cs

```diff
@@ -0,0 +1,30 @@
+﻿/*
+ * SonarScanner for .NET
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto: info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarScanner.MSBuild.PreProcessor;
+
+public abstract record class SonarServer(string ServerUrl)
+{
+    public string ServerUrl { get; } = ServerUrl;
```

Comment:
clarification: Wouldn't it be better to have this as `Uri` instead of string?

Reply:
I kept using a string. The Uri validation happens later when the HttpClient is constructed. But an Uri may help to identify "is smiliar" for these cases
* Scheme: `HTTP://sonarcloud.io`
* Login: `https://user@name:sonarcloud.io`
* Path: `https://sonarcloud.io/`
* Port: `https://sonarcloud.io:443/`
* Casing: `https://SonarCloud.io/`
and the like. Let's see what Julien is answering regarding this.